query/src/main/scala/filodb/query/exec/PromQlExec.scala

```diff
@@ -28,6 +28,9 @@ case class PromQlExec(id: String,
   protected def args: String = params.toString
   import PromQlExec._
 
+  val recSchema = SerializableRangeVector.toSchema(columns)
```

Comment:
recSchema can still be inside the companion object. It is immutable across queries and can be reused.

Reply:
Moved it to PromQlExec object